src/plugins/field_formats/common/converters/numeral.ts

```diff
@@ -36,6 +36,9 @@ export abstract class NumeralFormat extends FieldFormat {
     if (val === +Infinity) return '+∞';
     if (typeof val !== 'number') {
       val = parseFloat(val);
+      if (typeof val === 'object') {
```

Comment:
I think we should also extend types on L34? and add a test for it at `number.test.ts` since `NumberFormat extends NumeralFormat`